components/scream/src/dynamics/homme/interface/homme_driver_mod.F90

```diff
@@ -113,6 +119,19 @@ subroutine prim_copy_cxx_to_f90 (copy_phis)
     if (copy_phis) then
       ! Set phis=phi(bottom)
       elem_state_phis(:,:,:) = elem_state_phinh_i(:,:,nlevp,tl%n0,:)
+
+      ! Set geopotential views
+      elem_state_phis_local_ptr       = c_loc(elem_state_phis_local)
+      elem_derived_gradphis_local_ptr = c_loc(elem_gradphis_local)
+      do ie=1,nelemd
+        elem_state_phis_local = elem(ie)%state%phis
+        elem_gradphis_local   = elem(ie)%derived%gradphis
+        call init_geopotential_c (ie-1, elem_state_phis_local_ptr, elem_derived_gradphis_local_ptr)
```

Comment:
An alternative is to use `derivative_mod` here, and do
```
elem_gradphis_local = gradient_sphere(elem_state_phis_local,deriv,elem(ie)%Dinv)
```
so that you don't need to call `compute_gradphis_c` anymore (simply do the calculation in F90).

Reply:
Ok, I'll try that out.